src/query/src/log_query/planner.rs

```diff
@@ -0,0 +1,353 @@
+
+        // Build the final plan
+        let plan = plan_builder.build().context(DataFusionPlanningSnafu)?;
+
+        Ok(plan)
+    }
+
+    fn build_time_filter(&self, time_filter: &TimeFilter, schema: &Schema) -> Result<Expr> {
+        let timestamp_col = schema
+            .timestamp_column()
+            .with_context(|| TimeIndexNotFoundSnafu {})?
+            .name
+            .clone();
+
+        let expr = col(timestamp_col.clone())
+            .gt_eq(lit(ScalarValue::Utf8(time_filter.start.clone())))
+            .and(
+                col(timestamp_col).lt_eq(lit(ScalarValue::Utf8(
+                    time_filter
+                        .end
+                        .clone()
+                        .or(Some("9999-12-31T23:59:59Z".to_string())),
+                ))),
+            );
+
+        Ok(expr)
+    }
+
+    /// Returns filter expressions
+    fn build_column_filter(&self, column_filter: &ColumnFilters) -> Result<Option<Expr>> {
+        if column_filter.filters.is_empty() {
+            return Ok(None);
+        }
+
+        let exprs = column_filter
+            .filters
+            .iter()
+            .map(|filter| match filter {
+                log_query::ContentFilter::Exact(pattern) => Ok(col(&column_filter.column_name)
+                    .like(lit(ScalarValue::Utf8(Some(escape_pattern(pattern)))))),
+                log_query::ContentFilter::Prefix(pattern) => Ok(col(&column_filter.column_name)
+                    .like(lit(ScalarValue::Utf8(Some(format!(
+                        "{}%",
+                        escape_pattern(pattern)
+                    )))))),
+                log_query::ContentFilter::Postfix(pattern) => Ok(col(&column_filter.column_name)
+                    .like(lit(ScalarValue::Utf8(Some(format!(
+                        "%{}",
+                        escape_pattern(pattern)
+                    )))))),
+                log_query::ContentFilter::Contains(pattern) => Ok(col(&column_filter.column_name)
+                    .like(lit(ScalarValue::Utf8(Some(format!(
+                        "%{}%",
+                        escape_pattern(pattern)
+                    )))))),
+                log_query::ContentFilter::Regex(..) => Err::<Expr, _>(
+                    UnimplementedSnafu {
+                        feature: "regex filter",
+                    }
+                    .build(),
+                ),
+                log_query::ContentFilter::Compound(..) => Err::<Expr, _>(
+                    UnimplementedSnafu {
+                        feature: "compound filter",
+                    }
+                    .build(),
+                ),
+            })
+            .try_collect::<Vec<_>>()?;
+
+        Ok(conjunction(exprs))
+    }
+}
+
+fn escape_pattern(pattern: &str) -> String {
+    pattern
+        .chars()
+        .flat_map(|c| match c {
+            '\\' | '%' | '_' => vec!['\\', c],
+            _ => vec![c],
+        })
+        .collect::<String>()
+}
```

Comment:
Looks like the function is the exact same as the one in `common-function`'s `PatternAst::escape_pattern`. Can we use import instead?